speechbrain/dataio/encoder.py

```diff
@@ -655,6 +662,74 @@ def load_if_possible(self, path, end_of_epoch=False, device=None):
             return False
         return True  # If here, all good
 
+    def expect_len(self, expected_len):
+        """Specify the expected category count. If the category count observed
+        during encoding/decoding does NOT match this, an error will be raised.
+
+        This can prove useful to detect bugs in scenarios where the encoder is
+        dynamically built using a dataset, but downstream code expects a
+        specific category count (and may silently break otherwise).
+
+        This can be called anytime and the category count check will only be
+        performed during an actual encoding/decoding task.
+
+        Arguments
+        ---------
+        expected_len : int
+            The expected final category count, i.e. `len(encoder)`.
+
+        Example
+        -------
+        >>> encoder = CategoricalEncoder()
+        >>> encoder.update_from_iterable("abcd")
+        >>> encoder.expect_len(3)
+        >>> encoder.encode_label("a")
+        Traceback (most recent call last):
+          ...
+        RuntimeError: .expect_len(3) was called, but 4 categories found
+        >>> encoder.expect_len(4)
+        >>> encoder.encode_label("a")
+        0
+        """
+        # TODO
```

Comment:
?

Reply:
I think this is just a forgotten comment which can be removed.
```suggestion
```